dxr/plugins/rust/tests/test_data/test_fields.py

```diff
@@ -0,0 +1,153 @@
+                            [("let b = SomeFields { field1: 42, <b>field2</b>: NoFields };", 26),
+                             ("let _ = b.<b>field2</b>;", 28),
+                             ("let SomeFields { field1: b1, <b>field2</b>: b2 } = b;", 29),
+                             ("let SomeFields { field1, <b>field2</b> } = b;", 30),
+                             ("let c = SFAlias { field1: 42, <b>field2</b>: NoFields };", 32),
+                             ("let _ = c.<b>field2</b>;", 34),
+                             ("let SFAlias { field1: c1, <b>field2</b>: c2 } = c;", 35),
+                             ("let SFAlias { field1, <b>field2</b> } = c;", 36),
+                             ("EnumFields::Struct{ field1, <b>field2</b> } =&gt; {}", 42),
+                             ("EnumFields::Nested(SomeFields { field1, <b>field2</b> }) =&gt; {}", 43)])
+
+    # TODO case insensitive
+    def test_qual_field_case_insensitive(self):
+        self.found_line_eq('+var:test::SomeFields::field1', "<b>field1</b>: i32,", 9, is_case_sensitive=False)
+        self.found_line_eq('+var:test::SomeFields::field2', "<b>field2</b>: NoFields,", 10, is_case_sensitive=False)
+
+    def test_qual_enum_field_case_insensitive(self):
+        self.found_line_eq('+var:test::EnumFields::Struct::field1', "Struct{ <b>field1</b>: i32,", 18, is_case_sensitive=False)
+        self.found_line_eq('+var:test::EnumFields::Struct::field2', "<b>field2</b>: NoFields },", 19, is_case_sensitive=False)
+
+    def test_field_case_insensitive(self):
+        # Note the apparent field refs here, these are actually new variables
+        # being brought into scope.
+        self.found_lines_eq('var:field1', [("<b>field1</b>: i32,", 9),
+                                           ("Struct{ <b>field1</b>: i32,", 18),
+                                           ("let SomeFields { <b>field1</b>, field2 } = b;", 30),
+                                           ("let SFAlias { <b>field1</b>, field2 } = c;", 36),
+                                           ("EnumFields::Struct{ <b>field1</b>, field2 } =&gt; {}", 42),
+                                           ("EnumFields::Nested(SomeFields { <b>field1</b>, field2 }) =&gt; {}", 43)], is_case_sensitive=False)
+        self.found_lines_eq('var:field2', [("<b>field2</b>: NoFields,", 10),
+                                           ("<b>field2</b>: NoFields },", 19),
+                                           ("let SomeFields { field1, <b>field2</b> } = b;", 30),
+                                           ("let SFAlias { field1, <b>field2</b> } = c;", 36),
+                                           ("EnumFields::Struct{ field1, <b>field2</b> } =&gt; {}", 42),
+                                           ("EnumFields::Nested(SomeFields { field1, <b>field2</b> }) =&gt; {}", 43)], is_case_sensitive=False)
+
+    def test_qual_field_ref_case_insensitive(self):
+        self.found_lines_eq('+var-ref:test::SomeFields::field1',
```

Comment:
My thinking was that it is possible to break case senstitivity (e.g., by not using Qualified...), so we should test that it works. I probably went a bit overboard, just one or two tests would be OK, rather than loads.